hydra-node/src/Hydra/Options.hs

```diff
@@ -285,3 +414,95 @@ parseHydraOptions = getArgs <&> parseHydraOptionsFromString >>= handleParseResul
+
+-- | Convert an 'Options' instance into the corresponding list of command-line arguments.
+--
+-- This is useful in situations where one wants to programatically define 'Options', providing
+-- some measure of type safety, without having to juggle with strings.
+toArgs :: Options -> [String]
+toArgs
+  Options
+    { verbosity
+    , nodeId
+    , host
+    , port
+    , peers
+    , apiHost
+    , apiPort
+    , monitoringPort
+    , hydraSigningKey
+    , hydraVerificationKeys
+    , chainConfig
+    , ledgerConfig
+    } =
+    isVerbose verbosity
+      <> ["--node-id", show nodeId]
+      <> ["--host", show host]
+      <> ["--port", show port]
+      <> ["--api-host", show apiHost]
+      <> ["--api-port", show apiPort]
+      <> ["--hydra-signing-key", hydraSigningKey]
+      <> concatMap (\vk -> ["--hydra-verification-key", vk]) hydraVerificationKeys
+      <> concatMap toArgPeer peers
+      <> maybe [] (\mport -> ["--monitoring-port", show mport]) monitoringPort
+      <> argsChainConfig
+      <> argsLedgerConfig
+   where
+    isVerbose = \case
+      Quiet -> ["--quiet"]
+      _ -> []
+
+    toArgPeer p =
+      ["--peer", show p]
+
+    toArgStartChainFrom = \case
+      Just ChainPointAtGenesis ->
+        error "ChainPointAtGenesis"
+      Just (ChainPoint (SlotNo slotNo) headerHash) ->
+        let headerHashBase16 = toString (serialiseToRawBytesHexText headerHash)
+         in ["--start-chain-from", show slotNo <> "." <> headerHashBase16]
+      Nothing ->
+        []
+
+    toArgNetworkId = \case
+      Mainnet -> error "Mainnet not supported"
+      Testnet (NetworkMagic magic) -> show magic
+
+    argsChainConfig =
+      mempty
+        <> ["--network-id", toArgNetworkId networkId]
+        <> ["--node-socket", nodeSocket]
+        <> ["--cardano-signing-key", cardanoSigningKey]
+        <> concatMap (\vk -> ["--cardano-verification-key", vk]) cardanoVerificationKeys
+        <> toArgStartChainFrom startChainFrom
+
+    argsLedgerConfig =
+      mempty
+        <> ["--ledger-genesis", cardanoLedgerGenesisFile]
+        <> ["--ledger-protocol-parameters", cardanoLedgerProtocolParametersFile]
+
+    CardanoLedgerConfig
+      { cardanoLedgerGenesisFile
+      , cardanoLedgerProtocolParametersFile
+      } = ledgerConfig
+
+    DirectChainConfig
+      { networkId
+      , nodeSocket
+      , cardanoSigningKey
+      , cardanoVerificationKeys
+      , startChainFrom
+      } = chainConfig
+
+genFilePath :: String -> Gen FilePath
+genFilePath extension = do
+  path <- reasonablySized (listOf1 (elements ["a", "b", "c"]))
+  pure $ intercalate "/" path <> "." <> extension
+
+genChainPoint :: Gen ChainPoint
+genChainPoint = ChainPoint <$> (SlotNo <$> arbitrary) <*> someHeaderHash
+ where
+  someHeaderHash = do
+    bytes <- vectorOf 32 arbitrary
+    let hash = fromMaybe (error "invalid bytes") $ deserialiseFromRawBytes (proxyToAsType Proxy) . BS.pack $ bytes
+    pure hash
```

Comment:
Should: Move this to our `hydra-cardano-api` extension package. It feels a bit out of place here and `ChainPoint` is a `cardano-api` type.